lib/grape/validations/validators/coerce.rb

```diff
@@ -80,6 +76,13 @@ def coerce_value(val)
         Types::InvalidValue.new
       end
 
+      def coerce_nil(val)
+        return []      if type == Array
+        return Set.new if type == Set
+        return {}      if type == Hash
+        return val
```

Comment:
just `val`

Reply:
this is a good place for a `case`

```ruby
case type
when Array
when ...
```

Reply:
That doesn't actually work in this case because we need to distinguish between `type.is_a?(Array)` and `type == Array` (https://stackoverflow.com/questions/3908380/ruby-class-types-and-case-statements).